Automattic/jetpack
# Add QueryProducts Data Component

#### Changes proposed in this Pull Request:

This adds a `QueryProducts` data component along with the corresponding Redux code to provide price data to the Singe Product Jetpack Backup project.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
See the P2 here for the design of this PR: p1HpG7-7MK-p2 (specifically the part titled "WP Admin Desktop")
See the P2 here for the overall MT: p1HpG7-7ET-p2

#### Testing instructions:
* On your Jetpack site, go to `/wp-admin/admin.php?page=jetpack#/plans`
* Open dev tools and inspect the network tab. Verify that a call to `/jetpack/v4/products` was successful.
* Open Redux dev tools and inspect the actions. Verify that there is a JETPACK_PRODUCTS_FETCH action and a JETPACK_PRODUCTS_FETCH_RECEIVE action. 
* In the Redux state tree, visit jetpack > siteData > data > products and verify that the object there is correct.

#### Proposed changelog entry for your changes:

* None needed, should be covered under the overall changelog entry for the Single Product Backup changes.
